filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/stream/config/AkkaRunnableGraph.java

```diff
@@ -354,4 +343,65 @@ AuditServiceSinkFactory auditServiceSinkFactory(
                         .map(committableTokenOffset -> (Committable) committableTokenOffset.t1())
                         .to(committerSink));
     }
+
+    @Bean
+    RunnableGraph<Control> tokenErrorMessageRunnableGraph(final Source<CommittableMessage<String, AuditErrorMessage>, Control> tokenErrorMessageSource,
+                                                               final Sink<Committable, CompletionStage<Done>> committerSink,
+                                                               final ErrorMessageEventService errorMessageEventService) {
+        return tokenErrorMessageSource
+                // Extract committable, token and message
+                .map(committableMessage -> Tuple3.create(
+                        committableMessage.committableOffset(),
+                        new String(committableMessage.record().headers().lastHeader(Token.HEADER).value(), Charset.defaultCharset()),
+                        committableMessage.record().value()
+                ))
+
+                // Filter out anything from this service
+                .filter(auditErrorMessage -> !auditErrorMessage.t3().getServiceName().equals(SERVICE_NAME))
+
+                // Write exception to persistence
+                .mapAsync(PARALLELISM, committableTokenErrorMessage -> errorMessageEventService
+                        .putAuditErrorMessage(committableTokenErrorMessage.t2(), committableTokenErrorMessage.t3())
+                        .thenApply(ignored -> committableTokenErrorMessage))
+
+                // Commit processed message to kafka
+                .to(Flow.<Tuple3<CommittableOffset, String, AuditErrorMessage>>create()
+                        .map(committableTokenOffset -> (Committable) committableTokenOffset.t1())
```

Comment:
Can this not be
```suggestion
                        .<Committable>map(Tuple3::t1)
```

Reply:
Even better, could you not have the `mapAsync` return the committable and then have this just to the committerSink?

Reply:
updated